### PR TITLE
Translating ZH for Contributing document (zh review requested)

### PR DESCRIPTION
**Pull request by HelloSeaNation**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [done ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [done ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [done ] Include author(s) and platform where appropriate.
- [ n/a] Put lists in alphabetical order, correct spacing.
- [n/a ] Add needed indications (PDF, access notes, under construction).
- [done ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by HelloSeaNation** [comment]
commit translation

**Comment by eshellman** [comment]
will ask for help with review

**Comment by HelloSeaNation** [comment]
commit

**Comment by chenqianhe** [comment]
I'm not sure if you want to translate the document into Simplified Chinese or Traditional Chinese. I saw that some of the documents were translated from Traditional Chinese to Simplified Chinese, while others were translated from English to Traditional Chinese. Personally, I would prefer to provide Simplified Chinese; Alternatively, if possible, provide both separately. After determining the target language, I can provide some reviews

**Comment by HelloSeaNation** [comment]
I will make this as a simplified Chinese, and will create another with Tranditional Chinese with TW at the end of the file

**Comment by HelloSeaNation** [comment]
Correction is made, please review

File: docs/CONTRIBUTING-zh.md [hunk 1]
```diff
@@ -175,120 +175,116 @@
...
-##### URLs
+##### 网址
 
-- We don't permit shortened URLs.
-- Tracking codes must be removed from the URL.
-- International URLs should be escaped. Browser bars typically render these to Unicode, but use copy and paste, please.
-- Secure (`https`) URLs are always preferred over non-secure (`http`) urls where HTTPS has been implemented.
-- We don't like URLs that point to webpages that don't host the listed resource, but instead point elsewhere.
+- 我们不允许使用缩短的网址。
+- 必须从网址中删除追踪程式码。
```
Comment (Colin-coder) [inline docs/CONTRIBUTING-zh.md:204]:
```suggestion
- 必须从网址中删除追踪代码。
```
"程式码"是繁体中文使用的术语，简体中文建议修改为"代码"

File: docs/CONTRIBUTING-zh.md [hunk 2]
```diff
@@ -175,120 +175,116 @@
...
 
-##### URLs
+##### 网址
 
-- We don't permit shortened URLs.
-- Tracking codes must be removed from the URL.
-- International URLs should be escaped. Browser bars typically render these to Unicode, but use copy and paste, please.
-- Secure (`https`) URLs are always preferred over non-secure (`http`) urls where HTTPS has been implemented.
-- We don't like URLs that point to webpages that don't host the listed resource, but instead point elsewhere.
+- 我们不允许使用缩短的网址。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:203]: 我们不允许使用短链。

Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:203]:
```suggestion
- 我们不允许使用短链。
```

File: docs/CONTRIBUTING-zh.md [hunk 3]
```diff
@@ -175,120 +175,116 @@
...
 
-##### Titles
+##### 标题
 
-- No invented titles. We try to take titles from the resources themselves; contributors are admonished not to invent titles or use them editorially if this can be avoided. An exception is for older works; if they are primarily of historical interest, a year in parentheses appended to the title helps users know if they are of interest.
-- No ALLCAPS titles. Usually title case is appropriate, but when doubt use the capitalization from the source
-- No emojis.
+- 没有发明的标题。我们尝试从资源本身取得标题；告诫撰稿人不要发明标题或在编辑中使用它们，如果可以避免的话。较旧的作品除外；如果它们主要具有历史意义，标题后面括号中的年份可以帮助使用者知道它们是否感兴趣。
+- 没有全大写标题。通常标题大小写是合适的，但如有疑问，请使用来源中的大写
+- 没有表情符号。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:198]:
```suggestion
- 不要使用表情符号
```

File: docs/CONTRIBUTING-zh.md [hunk 4]
```diff
@@ -175,120 +175,116 @@
...
 
 
-##### Titles
+##### 标题
 
-- No invented titles. We try to take titles from the resources themselves; contributors are admonished not to invent titles or use them editorially if this can be avoided. An exception is for older works; if they are primarily of historical interest, a year in parentheses appended to the title helps users know if they are of interest.
-- No ALLCAPS titles. Usually title case is appropriate, but when doubt use the capitalization from the source
-- No emojis.
+- 没有发明的标题。我们尝试从资源本身取得标题；告诫撰稿人不要发明标题或在编辑中使用它们，如果可以避免的话。较旧的作品除外；如果它们主要具有历史意义，标题后面括号中的年份可以帮助使用者知道它们是否感兴趣。
+- 没有全大写标题。通常标题大小写是合适的，但如有疑问，请使用来源中的大写
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:197]:
```suggestion
- 不要使用全大写标题。常规的标题大小写是合适的，但如有疑问，请使用来源的大小写格式。
```

File: docs/CONTRIBUTING-zh.md [hunk 5]
```diff
@@ -175,120 +175,116 @@
...
+我们的清单提供了一组最小的元资料：标题、URL、创建者、平台和存取注释。
 
 
-##### Titles
+##### 标题
 
-- No invented titles. We try to take titles from the resources themselves; contributors are admonished not to invent titles or use them editorially if this can be avoided. An exception is for older works; if they are primarily of historical interest, a year in parentheses appended to the title helps users know if they are of interest.
-- No ALLCAPS titles. Usually title case is appropriate, but when doubt use the capitalization from the source
-- No emojis.
+- 没有发明的标题。我们尝试从资源本身取得标题；告诫撰稿人不要发明标题或在编辑中使用它们，如果可以避免的话。较旧的作品除外；如果它们主要具有历史意义，标题后面括号中的年份可以帮助使用者知道它们是否感兴趣。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:196]:
```suggestion
- 不要创作标题。我们尽量从资源本身获取标题；强烈建议贡献者不要创作标题，或者在可以避免的情况下不要编辑它们原始标题。对于较旧的作品有一个例外；如果它们主要具有历史价值，将年份括在标题后的括号内可以帮助用户了解它们是否有兴趣。
```

File: docs/CONTRIBUTING-zh.md [hunk 6]
```diff
@@ -175,120 +175,116 @@
...
+##### 网址
 
-- We don't permit shortened URLs.
-- Tracking codes must be removed from the URL.
-- International URLs should be escaped. Browser bars typically render these to Unicode, but use copy and paste, please.
-- Secure (`https`) URLs are always preferred over non-secure (`http`) urls where HTTPS has been implemented.
-- We don't like URLs that point to webpages that don't host the listed resource, but instead point elsewhere.
+- 我们不允许使用缩短的网址。
+- 必须从网址中删除追踪程式码。
+- 国际网址应进行转义。浏览器列通常会将这些内容呈现为 Unicode，但请使用复制和贴上。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:205]: 贴上 -> 粘贴

File: docs/CONTRIBUTING-zh.md [hunk 7]
```diff
@@ -175,120 +175,116 @@
...
+- 安全性 (`https`) URL 始终优先于已实作 HTTPS 的非安全性 (`http`)网址。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:206]:
```suggestion
- 在支持 HTTPS 的网站，安全的（https）URL总是优于非安全的（http）URL。
```

File: docs/CONTRIBUTING-zh.md [hunk 8]
```diff
@@ -175,120 +175,116 @@
...
+##### 创作者
 
-- We want to credit the creators of free resources where appropriate, including translators!
-- For translated works the original author should be credited. We recommend using [MARC relators](https://loc.gov/marc/relators/relaterm.html) to credit creators other than authors, as in this example:
+- 我们希望在适当的情况下赞扬免费资源的创建者，包括翻译人员！
+- 翻译作品，应注明原作者。我们建议使用 [MARC relators](https://loc.gov/marc/relators/relaterm.html) 来表彰作者以外的创作者，如下例所示：
 
     ```markdown
-    * [A Translated Book](http://example.com/book-zh.html) - John Doe, `trl.:` Mike The Translator
+    * [一本翻译书](http://example.com/book-zh.html) - John Doe，`trl.:` 翻译者麦克
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:216]:
```suggestion
    * [译作](http://example.com/book-zh.html) - John Doe，`trl.:` 译者麦克
```

File: docs/CONTRIBUTING-zh.md [hunk 9]
```diff
@@ -175,120 +175,116 @@
...
     ```
+    这里，注解「trl.:」使用「翻译器」的 MARC 相关程式码。
+- 使用逗号「,」来分隔作者清单中的每个项目。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:219]:
```suggestion
- 使用逗号「，」来分隔作者清单中的每个项目。
```

I found you use ' ，' on md example.

File: docs/CONTRIBUTING-zh.md [hunk 10]
```diff
@@ -175,120 +175,116 @@
...
+- 您可以使用「`et al.`」来缩短作者清单。
+- 我们不允许创作者连结。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:221]:
```suggestion
- 我们不允许为创作者提供链接。
```

I am not sure about this

File: docs/CONTRIBUTING-zh.md [hunk 11]
```diff
@@ -175,120 +175,116 @@
...
-    here, the annotation `trl.:` uses the MARC relator code for "translator".
-- Use a comma `,` to delimit each item in the author list.
-- You can shorten author lists with "`et al.`".
-- We do not permit links for Creators.
-- For compilation or remixed works, the "creator" may need a description. For example, "GoalKicker" or "RIP Tutorial" books are credited as "`Compiled from StackOverflow documentation`".
 
+##### 平台及接入注意事项
 
-##### Platforms and Access Notes
+- 培训班。特别是对于我们的课程清单来说，平台是资源描述的重要组成部分。这是因为课程平台具有不同的可供性和可访问性模式。虽然我们通常不会列出需要注册的书籍，但许多课程平台都有一些功能，如果没有某种帐户就无法使用。范例课程平台包括 Coursera、EdX、Udacity 和 Udemy。当课程依赖平台时，平台名称应列在括号中。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:227]:
```suggestion
- 课程。尤其是对于我们的课程列表，平台是资源描述的重要部分。这是因为不同的课程平台有不同的功能和访问模型。尽管我们通常不会列出需要注册的书籍，但许多课程平台的功能在没有某种帐户的情况下无法工作。课程平台的例子包括 Coursera、EdX、Udacity 和 Udemy。当一个课程依赖于一个平台时，应在括号中列出平台名称。
```

File: docs/CONTRIBUTING-zh.md [hunk 12]
```diff
@@ -175,120 +175,116 @@
...
+- Youtube.我们有许多包含 YouTube 播放清单的课程。我们不会将 YouTube 列为平台，而是尝试列出 YouTube 创作者，这通常是一个子平台。
+- YouTube影片。我们通常不会连结到单一 YouTube 视频，除非它们的长度超过一个小时并且结构类似于课程或教程。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:229]:
```suggestion
- YouTube 影片。我们通常不会链接到单一 YouTube 视频，除非它们的长度超过一个小时并且结构类似于课程或教程。
```

File: docs/CONTRIBUTING-zh.md [hunk 13]
```diff
@@ -175,120 +175,116 @@
...
 
-- Courses. Especially for our course lists, the platform is an important part of the resource description. This is because course platforms have different affordances and access models. While we usually won't list a book that requires a registration, many course platforms have affordances that don't work without some sort of account. Example course platforms include Coursera, EdX, Udacity, and Udemy. When a course depends on a platform, the platform name should be listed in parentheses.
-- YouTube. We have many courses which consist of YouTube playlists. We do not list YouTube as a platform, we try to list the YouTube creator, which is often a sub-platform.
-- YouTube videos. We usually don't link to individual YouTube videos unless they are more than an hour long and are structured like a course or a tutorial.
-- Leanpub. Leanpub hosts books with a variety of access models. Sometimes a book can be read without registration; sometimes a book requires a Leanpub account for free access. Given quality of the books and the mixture and fluidity of Leanpub access models, we permit listing of the latter with the access note `*(Leanpub account or valid email requested)*`.
+#### 流派
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:232]: Matbe 体裁 or 类型 is better. I am sure 流派 is not good

File: docs/CONTRIBUTING-zh.md [hunk 14]
```diff
@@ -175,120 +175,116 @@
...
 
+决定资源属于哪个清单的第一条规则是查看资源如何描述自己。如果它称自己为一本书，那么也许它就是一本书。
 
-#### Genres
+##### 我们未列出的流派
 
-The first rule in deciding which list a resource belongs in is to see how the resource describes itself. If it calls itself a book, then maybe it's a book.
+由于互联网非常庞大，因此我们不将以下内容包括在列表中：
 
+- 部落格文章
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:240]:
```suggestion
- 博客（博客网站）
```

File: docs/CONTRIBUTING-zh.md [hunk 15]
```diff
@@ -175,120 +175,116 @@
...
+- 文章
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:241]:
```suggestion
- 博客文章
```

File: docs/CONTRIBUTING-zh.md [hunk 16]
```diff
@@ -175,120 +175,116 @@
...
+- 网站（那些托管我们列出的大量项目的网站除外）。
+- 不是课程或截图影片的影片。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:243]:
```suggestion
- 不属于课程或录屏的视频。
```

File: docs/CONTRIBUTING-zh.md [hunk 17]
```diff
@@ -175,120 +175,116 @@
...
+- 书籍章节
+- 书中的预告片样本
+- IRC 或 Telegram 频道
+- Slack或邮件列表
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:247]:
```suggestion
- Slack 或邮件列表
```

File: docs/CONTRIBUTING-zh.md [hunk 18]
```diff
@@ -175,120 +175,116 @@
...
 
-The first rule in deciding which list a resource belongs in is to see how the resource describes itself. If it calls itself a book, then maybe it's a book.
+由于互联网非常庞大，因此我们不将以下内容包括在列表中：
 
+- 部落格文章
+- 文章
+- 网站（那些托管我们列出的大量项目的网站除外）。
+- 不是课程或截图影片的影片。
+- 书籍章节
+- 书中的预告片样本
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:245]:
```suggestion
- 书籍的预览样章
```

File: docs/CONTRIBUTING-zh.md [hunk 19]
```diff
@@ -175,120 +175,116 @@
...
+- IRC 或 Telegram 频道
+- Slack或邮件列表
 
-##### Genres we don't list
+我们的竞争性节目清单对这些例外情况没有那么严格。 repo的范围由社区决定；如果您想建议更改或新增范围，请使用问题提出建议
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:249]:
```suggestion
我们的竞争性节目清单对这些例外情况没有那么严格。 仓库的范围由社区决定；如果您想建议更改或新增范围，请使用 issue 提出建议
```

I am not sure what is the mean of competitive programming lists

File: docs/CONTRIBUTING-zh.md [hunk 20]
```diff
@@ -175,120 +175,116 @@
...
-Courses often have associated textbooks, which we would list in our books lists. Courses have lectures, exercises, tests, notes or other didactic aids. A single lecture or video by itself is not a course. A powerpoint is not a course.
+如果您可以将其打印出来并保留其精髓，那么它就不是交互式教程。
 
 
-##### Interactive Tutorials vs. Other stuff
+### 自动化
 
-If you can print it out and retain its essence, it's not an Interactive Tutorial.
+- 格式化规则的执行是通过以下方式自动执行的 [GitHub Actions](https://github.com/features/actions) 用 [fpb-lint](https://github.com/vhf/free-programming-books-lint) (see [`.github/workflows/fpb-lint.yml`](../.github/workflows/fpb-lint.yml))
+- URL 验证用途 [awesome_bot](https://github.com/dkhamsing/awesome_bot)
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:282]:
```suggestion
- URL 验证使用 [awesome_bot](https://github.com/dkhamsing/awesome_bot)
```

File: docs/CONTRIBUTING-zh.md [hunk 21]
```diff
@@ -175,120 +175,116 @@
...
+- 要触发 URL 验证，请推送包含以下内容的提交消息的提交 `check_urls=file_to_check`:
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:283]:
```suggestion
- 要触发 URL 验证，请推送包含以下内容的 commit message 的提交 `check_urls=file_to_check`:
```

File: docs/CONTRIBUTING-zh.md [hunk 22]
```diff
@@ -175,120 +175,116 @@
...
-    ```properties
+    ```特性
     check_urls=free-programming-books.md free-programming-books-zh.md
     ```
-
-- You may specify more than one file to check, using a single space to separate each entry.
-- If you specify more than one file, results of the build are based on the result of the last file checked. You should be aware that you may get passing green builds due to this so be sure to inspect the build log at the end of the Pull Request by clicking on "Show all checks" -> "Details".
+    
+- - 您可以指定多个要检查的文件，使用单个空格分隔每个条目。
+- 如果指定多个文件，则构建结果基于最后检查的文件的结果。您应该意识到，您可能会因此而通过绿色构建，因此请务必通过单击“Show all checks”->“Details”来检查拉取请求末尾的构建日志。
```
Comment (chenqianhe) [inline docs/CONTRIBUTING-zh.md:290]:
```suggestion
- 如果您指定了多个文件，构建的结果将基于最后一个检查的文件的结果。您应该注意，由于这个原因，您可能会得到通过的绿色构建，所以请确保在拉取请求结束时检查构建日志，点击 “Show all checks”->“Details”。
```

File: docs/CONTRIBUTING-zh.md [hunk 23]
```diff
@@ -175,120 +175,116 @@
 
 ### 按字母順序
```
Comment (SkywalkerDarren) [inline docs/CONTRIBUTING-zh.md:176]: 改成：按字母顺序

**Review by SkywalkerDarren — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md [hunk 24]
```diff
@@ -1053,7 +1053,6 @@
 
 * [Beginner's Typescript](https://www.totaltypescript.com/tutorials/beginners-typescript) - Matt Pocock
 * [ES6 and Typescript Tutorial](https://www.youtube.com/playlist?list=PLC3y8-rFHvwhI0V5mE9Vu6Nm-nap8EcjV) - Codevolution, Vishwas Gopinath
-* [Introduction to TypeScript](https://www.udemy.com/typescript/) - Daniel Stern (Udemy)
```
Comment (eshellman) [inline courses/free-courses-en.md:1056]: not sure why this is here

**Comment by eshellman** [comment]
thank you @chenqianhe @Colin-coder and @SkywalkerDarren for help with reviews. Please resolve the suggestions and mark LGTM so I can merge them.
Our lists should include both simplified and traditional in on file, but if we have versions of the documents in both, they should be labelled zh_CN and zh_TW and presented side-by-side in the docs index.

File: courses/free-courses-en.md (hunk 24 above)
Comment (HelloSeaNation) [reply courses/free-courses-en.md:1056]: Sorry, I made a mistake here, I will correct this file.

File: docs/CONTRIBUTING-zh.md (hunk 23 above)
Comment (HelloSeaNation) [reply docs/CONTRIBUTING-zh.md:176]: What is the different between the text I have created and compare to you?

File: docs/CONTRIBUTING-zh.md (hunk 13 above)
Comment (HelloSeaNation) [reply docs/CONTRIBUTING-zh.md:232]: I think 类型 is better

File: docs/CONTRIBUTING-zh.md [hunk 25]
```diff
@@ -175,97 +175,95 @@
...
 
-- 當有多個以相同字母開頭的標題時，按第二個排序，依此類推。例如：“aa”位於 “ab” 之前
-- “onetwo” 位於 “onetwo”之前
+- 当有多个以相同字母开头的标题时，按第二个排序，依此类推。例如：“aa”位于 “ab” 之前
+- “onetwo” 位于 “onetwo”之前
 
-如果您看到錯誤的鏈接，請檢查 linter 錯誤訊息以了解應該交換哪些行。
+如果您看到错误的链接，请检查 linter 错误讯息以了解应该交换哪些行。
```
[the remark this answers is not in the document]
Comment (HelloSeaNation) [reply docs/CONTRIBUTING-zh.md:181]: done

File: docs/CONTRIBUTING-zh.md (hunk 2 above)
Comment (HelloSeaNation) [reply docs/CONTRIBUTING-zh.md]: done

File: docs/CONTRIBUTING-zh.md (hunk 6 above)
Comment (HelloSeaNation) [reply docs/CONTRIBUTING-zh.md:205]: done, thanks!

File: docs/CONTRIBUTING-zh.md (hunk 23 above)
Comment (HelloSeaNation) [reply docs/CONTRIBUTING-zh.md]: What do think about "按字母顺序排列" ?

**Review by chenqianhe — APPROVED** [review]

**Review by overstarry — APPROVED** [review]

File: docs/CONTRIBUTING-zh.md (hunk 23 above)
Comment (SkywalkerDarren) [reply docs/CONTRIBUTING-zh.md]: “按字母顺序排列” is ok

File: docs/CONTRIBUTING-zh.md [hunk 26]
```diff
@@ -173,122 +173,118 @@
     正确: * [A Way-backed Interesting Book](https://web.archive.org/web/20211016123456/http://example.com/) - John Doe (HTML) *(:card_file_box: archived)*
     ```
 
-### 按字母順序
+### 按字母顺序
 
-- 當有多個以相同字母開頭的標題時，按第二個排序，依此類推。例如：“aa”位於 “ab” 之前
-- “onetwo” 位於 “onetwo”之前
+- 当有多个以相同字母开头的标题时，按第二个排序，依此类推。例如：“aa”位于 “ab” 之前
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:178]:
```suggestion
- 当有多个以相同字母开头的标题时，按第二个排序，依此类推。例如：“aa” 位于 “ab” 之前
```

File: docs/CONTRIBUTING-zh.md [hunk 27]
```diff
@@ -173,122 +173,118 @@
...
-- Use a comma `,` to delimit each item in the author list.
-- You can shorten author lists with "`et al.`".
-- We do not permit links for Creators.
-- For compilation or remixed works, the "creator" may need a description. For example, "GoalKicker" or "RIP Tutorial" books are credited as "`Compiled from StackOverflow documentation`".
 
+##### 平台及接入注意事项
 
-##### Platforms and Access Notes
+- 课程。尤其是对于我们的课程列表，平台是资源描述的重要部分。这是因为不同的课程平台有不同的功能和访问模型。尽管我们通常不会列出需要注册的书籍，但许多课程平台的功能在没有某种帐户的情况下无法工作。课程平台的例子包括 Coursera、EdX、Udacity 和 Udemy。当一个课程依赖于一个平台时，应在括号中列出平台名称。
+- Youtube.我们有许多包含 YouTube 播放清单的课程。我们不会将 YouTube 列为平台，而是尝试列出 YouTube 创作者，这通常是一个子平台。
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:228]:
```suggestion
- Youtube。我们有许多包含 YouTube 播放清单的课程。我们不会将 YouTube 列为平台，而是尝试列出 YouTube 创作者，这通常是一个子平台。
```

File: docs/CONTRIBUTING-zh.md [hunk 28]
```diff
@@ -173,122 +173,118 @@
...
+- YouTube 影片。我们通常不会链接到单一 YouTube 视频，除非它们的长度超过一个小时并且结构类似于课程或教程。
+- Leanpub。 Leanpub 托管具有多种存取模式的书籍。有时一本书无需注册即可阅读；有时一本书需要 Leanpub 帐户才能免费存取。考虑到书籍的品质以及 Leanpub 存取模型的混合性和流动性，我们允许使用存取注释「*（Leanpub 帐户或请求的有效电子邮件）*」列出后者。
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:230]:
```suggestion
- Leanpub。Leanpub 托管具有多种存取模式的书籍。有时一本书无需注册即可阅读；有时一本书需要 Leanpub 帐户才能免费存取。考虑到书籍的品质以及 Leanpub 存取模型的混合性和流动性，我们允许使用存取注释「*（Leanpub 帐户或请求的有效电子邮件）*」列出后者。
```

File: docs/CONTRIBUTING-zh.md [hunk 29]
```diff
@@ -173,122 +173,118 @@
...
 
-如果您看到錯誤的鏈接，請檢查 linter 錯誤訊息以了解應該交換哪些行。
+如果您看到错误的链接，请检查 linter 错误信息以了解应该交换哪些行。
 
 
-### 筆記
+### 笔记
 
-雖然基礎知識相對簡單，但我們列出的資源卻多種多樣。以下是關於我們如何處理這種多樣性的一些說明。
+虽然基础知识相对简单，但我们列出的资源却多种多样。以下是关于我们如何处理这种多样性的一些说明。
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:186]:
```suggestion
虽然基础知识相对简单，但我们列出的资源却多种多样。以下是关于我们如何处理这种多样性的一些说明：
```

这里使用冒号会更好

File: docs/CONTRIBUTING-zh.md [hunk 30]
```diff
@@ -173,122 +173,118 @@
...
-- To trigger URL validation, push a commit that includes a commit message containing `check_urls=file_to_check`:
-
-    ```properties
+    ```特性
     check_urls=free-programming-books.md free-programming-books-zh.md
     ```
-
-- You may specify more than one file to check, using a single space to separate each entry.
-- If you specify more than one file, results of the build are based on the result of the last file checked. You should be aware that you may get passing green builds due to this so be sure to inspect the build log at the end of the Pull Request by clicking on "Show all checks" -> "Details".
+
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:288]:
```suggestion

```

it is not necessary to keep empty space at the end of line

File: docs/CONTRIBUTING-zh.md [hunk 31]
```diff
@@ -173,122 +173,118 @@
...
 
-Courses often have associated textbooks, which we would list in our books lists. Courses have lectures, exercises, tests, notes or other didactic aids. A single lecture or video by itself is not a course. A powerpoint is not a course.
+如果您可以将其打印出来并保留其精髓，那么它就不是交互式教程。
 
 
-##### Interactive Tutorials vs. Other stuff
+### 自动化
 
-If you can print it out and retain its essence, it's not an Interactive Tutorial.
+- 格式化规则的执行是通过以下方式自动执行的 [GitHub Actions](https://github.com/features/actions) 用 [fpb-lint](https://github.com/vhf/free-programming-books-lint) (see [`.github/workflows/fpb-lint.yml`](../.github/workflows/fpb-lint.yml))
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:281]:
```suggestion
- 格式化规则的执行是通过以下方式自动执行的 [GitHub Actions](https://github.com/features/actions) 用 [fpb-lint](https://github.com/vhf/free-programming-books-lint) (参见 [`.github/workflows/fpb-lint.yml`](../.github/workflows/fpb-lint.yml))
```

File: docs/CONTRIBUTING-zh.md [hunk 32]
```diff
@@ -173,122 +173,118 @@
...
+- URL 验证使用 [awesome_bot](https://github.com/dkhamsing/awesome_bot)
+- 要触发 URL 验证，请推送包含以下内容的 commit message 的提交 `check_urls=file_to_check`:
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:283]:
```suggestion
- 要触发 URL 验证，请推送包含以下内容的 commit message 的提交 `check_urls=file_to_check`：
```

File: docs/CONTRIBUTING-zh.md [hunk 33]
```diff
@@ -173,122 +173,118 @@
...
-- IRC or Telegram channels
-- Slacks or mailing lists
+##### 书籍与其他东西
 
-Our competitive programming lists are not as strict about these exclusions. The scope of the repo is determined by the community; if you want to suggest a change or addition to the scope, please use an issue to make the suggestion.
+我们对于书本的性质并不那么挑剔。以下是一些表明资源是一本书的特征：
 
+- 它有一个 ISBN（国际标准书号）
+- 它有一个目录
+- 提供可下载版本，尤其是 ePub 档案。
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:258]:
```suggestion
- 提供可下载版本，尤其是 ePub 档案
```

File: docs/CONTRIBUTING-zh.md [hunk 34]
```diff
@@ -173,122 +173,118 @@
...
+由于互联网非常庞大，因此我们不将以下内容包括在列表中：
 
+- 博客（博客网站）
+- 博客文章
+- 网站（那些托管我们列出的大量项目的网站除外）。
+- 不属于课程或录屏的视频。
+- 书籍章节
+- 书籍的预览样章
+- IRC 或 Telegram 频道
+- Slack 或邮件列表
```
Comment (skyclouds2001) [inline docs/CONTRIBUTING-zh.md:247]:
```suggestion
- 博客（博客网站）
- 博客文章
- 网站（那些托管我们列出的大量项目的网站除外）
- 不属于课程或录屏的视频
- 书籍章节
- 书籍的预览样章
- IRC 或 Telegram 频道
- Slack 或邮件列表
```

**Comment by eshellman** [comment]
nice collaboration!

**Pull request merged**